server/src/types.ts

```diff
@@ -2,25 +2,41 @@ import { Room } from './rooms'
 import { User, MinimalUser, PublicUser } from './user'
 import { RoomNote } from './roomNote'
 
+export interface HappeningNowEntry {
+  text: string,
+  roomId?: string,
+  externalLink?: string
+}
+
+function isHappeningNowEntry (obj: any): obj is HappeningNowEntry {
```

Comment:
I'm having this awkward slapfight with Typescript with regards to server-side validation that makes me feel like a right idiot. This doesn't do what I really want it to do but it's Good Enough I think...but there's something I'm completely not understanding.

Reply:
Yeah, I think there are ways this could be made a lot cleaner and do more what you're intending it to, but I think those changes also bubble up to refactoring how `toServerSettings` works. Happy to go back after the event and refactor stuff together if you're interested in learning/figuring stuff out, but also agree this is probably Good Enough for T-36 hours.